cns/service/main.go

```diff
@@ -316,6 +380,33 @@ func main() {
 		go httpRestService.SendNCSnapShotPeriodically(cnsconfig.TelemetrySettings.SnapshotIntervalInMins, stopSnapshots)
 	}
 
+	// If CNS is running on managed DNC mode
+	if privateEndpoint != "" && infravnet != "" && nodeID != "" {
```

Comment:
Let's not decide if we need to use mdnc based on privateEndpoint != "" && infravnet != "" && nodeID != ""
Let's add a flag in config and based on that decide. The logic becomes simpler and clear intent is provided to CNS from client